From c191cfe754de2c37602056bb90eb197b7e9762ed Mon Sep 17 00:00:00 2001
From: Agis Anastasopoulos <a@xz0.org>
Date: Fri, 22 Sep 2023 13:31:00 +0300
Subject: [PATCH] Support skipping loading .env file

This patch adds supports for skipping loading the `.env` file, by
setting the HIVEMIND_SKIP_ENV environment variable. For example:

    $ HIVEMIND_SKIP_ENV=1 hivemind

Resolves #28.
---
 main.go | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/main.go b/main.go
index 7c2bf58..7a9c8d7 100644
--- a/main.go
+++ b/main.go
@@ -6,7 +6,7 @@ import (
 
 	"github.com/urfave/cli"
 
-	_ "github.com/DarthSim/godotenv/autoload"
+	"github.com/DarthSim/godotenv"
 )
 
 const version = "1.1.0"
@@ -17,6 +17,11 @@ func main() {
 		err  error
 	)
 
+	_, skipEnv := os.LookupEnv("HIVEMIND_SKIP_ENV")
+	if !skipEnv {
+		godotenv.Load()
+	}
+
 	app := cli.NewApp()
 
 	app.Name = "Hivemind"